spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/openai/OpenAiAutoConfiguration.java

```diff
@@ -49,7 +56,19 @@ public OpenAiService theoOpenAiService(OpenAiProperties openAiProperties) {
 			throw new IllegalArgumentException(
```

Comment:
Would it make sense to only force an API key if the URL was anything other than the default base URL (maybe that's too complex or nondeterministic)?

Reply:
I've seen two cases of model hosting projects that expose the OpenAI API locally and do not require a key.  I think we can remove the requirement for a key when the base url is not `https://api.openai.com` and see how it goes.